experimental/devices/mfrc522/mfrc522.go

```diff
@@ -321,12 +350,28 @@ func (r *Dev) Request() (int, error) {
 
 // Wait wait for IRQ to strobe on the IRQ pin when the card is detected.
 func (r *Dev) Wait() error {
+	r.mu.Lock()
```

Comment:
You absolutely need to do:
```
select {
  case <- r.stop:
  default:
}
```

before, but even so there's a small risk of deadlock if multiple Halt() are called concurrently, but that's not really a supported use case so I think you're fine here.

Another option is to use a condition variable; https://golang.org/pkg/sync/#Cond

Actually, the more I look at your goroutines, the more I think a sync.Cond is the right locking object since you really want to pass events.